zcash_client_backend/src/data_api/testing.rs

```diff
@@ -1147,9 +1148,11 @@ pub struct InitialChainState {
 
 pub trait DataStoreFactory {
     type Error: core::fmt::Debug;
-    type AccountId: ConditionallySelectable + Default + Send + 'static;
-    type DataStore: InputSource<AccountId = Self::AccountId>
-        + WalletRead<AccountId = Self::AccountId>
+    type AccountId: ConditionallySelectable + Default + Hash + Eq + Send + 'static;
+    type Account: Account<AccountId = Self::AccountId> + Clone;
```

Comment:
Now that we've added `Account` as an associated type, do we need `AccountId` to also be an associated type? Or can we just use the one that `Account` also now has? My guess is that we need it here in order to apply the required bounds (because IIRC Rust can't yet do so on the `type Account: Account` type - though also IIRC that is going to be possible Soon™️).